book/visual-effects.md

```diff
@@ -1154,89 +1103,71 @@ def paint_tree(layout_object, display_list):
-returns another list of commands. It's just that the output list is
-always a single `SaveLayer` command that wraps the original
-content---which makes sense, because first we need to draw the
-commands to a surface, and *then* apply transparency to it when
-blending into the parent.
 
-::: {.further}
-[This blog post](https://ciechanow.ski/alpha-compositing/) gives a really nice
-visual overview of many of the same concepts explored in this chapter,
-plus way more content about how a library such as Skia might implement features
-like raster sampling of vector graphics for lines and text, and interpolation
-of surfaces when their pixel arrays don't match resolution or orientation. I
-highly recommend it.
-:::
 
 Compositing pixels
 ==================
 
-Now let's pause and explore how opacity actually works under the hood.
-Skia, SDL, and many other color libraries account for opacity with a
-fourth *alpha* value for each pixel.[^alpha-vs-opacity] An alpha of 0
-means the pixel is fully transparent (meaning, no matter what the
-colors are, you can't see them anyway), and an alpha of 1 means a
-fully opaque.
+Let's pause to discuss how opacity actually works under the hood. As
+mentioned before, Skia surfaces store an alpha value for each pixel in
+the surface.[^alpha-vs-opacity] Then, when one surface is drawn to
+another, overlapping pixels are paired up and the colors are blended
+together, as defined by the `Paint` object passed to `saveLayer`.
 
 [^alpha-vs-opacity]: The difference between opacity and alpha can be
 confusing. Think of opacity as a visual effect *applied to* content,
 but alpha as a *part of* content. Think of alpha as implementation
 technique for representing opacity.
 
-When a pixel with alpha overlaps another pixel, the final color is a
-mix of their two colors. How exactly the colors are mixed is defined
-by Skia's `Paint` objects. Of course, Skia is pretty complex, but we
-can sketch these paint operations in Python as methods on an imaginary
-`Pixel` class.
-
-``` {.python file=examples}
-class Pixel:
-    def __init__(self, r, g, b, a):
-        self.r = r
-        self.g = g
-        self.b = b
-        self.a = a
-```
-
-When we apply a `Paint` with an `Alphaf` parameter, the first thing
-Skia does is add the requested opacity to each pixel:
+So, for example, when we apply a `Paint` with an `Alphaf` parameter,
+the first thing Skia does is add the requested opacity to each pixel:
 
 ``` {.python file=examples}
 class Pixel:
     def alphaf(self, opacity):
         self.a = self.a * opacity
 ```
 
-I want to emphasize that this code is not a part of our browser---I'm
-simply using Python code to illustrate what Skia is doing internally.
+Again, this code is not a part of our browser---I'm simply using
+Python code to illustrate what Skia is doing internally. Specifically,
+this `alphaf` operation modifies the pixels in the temporary surface,
+usually called the "source" or "top" surface:
```

Comment:
Who uses "top"? I suggest we just say source and dest.

Reply:
Got it.